src/airbrakes.py

```diff
@@ -1,6 +1,190 @@
+import board
+import busio
+import adafruit_ads1x15.ads1115 as ADS
+from adafruit_ads1x15.analog_in import AnalogIn
+
+
+"""
+This class implements a Kalman filter to determine how far the airbrakes
+flaps should be deployed, based on sensor data and previous flight data.
+
+
+Attributes
+----------
+
+
+
+Methods
+-------
+
+
+
+
+"""
 class KalmanFilter:
   def __init__(self, rd):
     rocket_data = rd
+
+
+
+"""
+
+Airbrakes: This class provides a simple way to interface with the airbrakes
+hardware. It is used by the KalmanFilter class.
+
+
+Attributes
+----------
+
+__max_pot_val [double]: Value of the potentiometer as read by the ADC when brakes are
+fully deployed. Set by calibrate().
+
+__min_pot_val [double]: Value of the potentiometer as read by the ADC when brakes are
+fully closed. Set by calibrate().
+
+pot_pin [int]: ADS1115 pin that is connected to potentiometer
+step_pin [int]: GPIO pin that controls stepping of A4988 stepper driver
+dir_pin [int]: GPIO pin that controls direction of A4988 stepper driver
+
+motor_direction [bool]: This value is XOR'd with the direction passed to
+__singleStep(). For the user, this means if the flaps are opening in the wrong
+direction when testing, invert the value of motor_direction.
+
+step_delay [int]: Delay in seconds between pulses of step_pin
+step_angle [int]: Step angle in degrees of stepper motor
+microsteps [int]: 2 (half), 4 (quarter), 16 (sixteenth) etc.
+gear_ratio [int]: (# of motor turns) / (# of gearbox output turns)
+
+__max_steps_to_open [int]: Maximum number of steps needed to fully open the brakes.
+This is calculated from the motor parameters: step_angle, microsteps, gear_ratio
+
+
+Methods
+-------
+
+__singleStep():
+  Steps Airbrakes stepper motor once in specified direction.
+
+  Returns nothing.
+
+  Param: step_direction
+    Passing step_direction=True opens Airbrakes one step
+    Passing step_direction=False closes Airbrakes one step
+
+    Note: If passing True closes Airbrakes, invert the value of 
+    step_direction when initializing Airbrakes object
+
+calibrate(): 
+  Sets __max_pot_val by reading potentiometer in the fully open state
+  and sets __min_pot_val by reading the potentiometer in the closed state
+
+  Returns nothing.
+
+  Must be called before launch (on the launch pad) to calibrate the system.
+
+deployBrakes():
+  Opens the brakes to the specifed percentage of fully open.
+
+  Returns value of potentiometer at final position.
+
+  Note: This uses the potentiometer for feedback so there will be a small
+  error associated with this. Adjust this error by changing max_error in
+  method body.
+
+
+"""
+class Airbrakes:
+  
+  # Set when calibrate() is called
+  __max_pot_val = None
+  __min_pot_val = None
+
+  def __init__(self,
+      stepper_pin,
+      direction_pin, 
+      direction: bool,
+      ads_pot_pin=0,
+      step_angle=1.8,
+      microsteps=16,
+      step_delay_micro_sec=75,
+      gear_box_ratio=14):
```

Comment:
```suggestion
  def __init__(self,
                   stepper_pin,
                   direction_pin, 
                   direction: bool,
                   ads_pot_pin=0,
                   step_angle=1.8,
                   microsteps=16,
                   step_delay_micro_sec=75,
                   gear_box_ratio=14):
```

my tabbing suggestion was off lol